pkg/storage/ingest/pusher.go

```diff
@@ -34,6 +34,7 @@ type pusherConsumer struct {
 
 type parsedRecord struct {
 	*mimirpb.WriteRequest
+	ctx      context.Context
```

Comment:
Can you add a comment to explain what is this context, and what it is used for, please?